lib/secret-storage/secret-storage.c

```diff
@@ -227,8 +227,9 @@ gboolean
 secret_storage_subscribe_for_key(const gchar *key, SecretStorageCB func, gpointer user_data)
 {
 
+  gpointer *ignore_orig_key, *ignore_value;
   SecretStorage *secret_storage;
-  if (!g_hash_table_contains(secret_manager, key))
+  if (!g_hash_table_lookup_extended(secret_manager, key, ignore_orig_key, ignore_value))
```

Comment:
Changing g_hash_table_contains is fine for me. Though with g_hash_table_lookup_extended there is a problem here, ignore_orig_key and ignore_value should be output parameters, meaning you should have passed &ignore_orig_key and &ignore_value. With the current version, glib would write  data to random location based on the garbage picked from the stack.

Though I would suggest something else: you can safely use g_hash_table_lookup here. From architecture point of view, either when adding a secret, either someone registers for a secret, we will allocate a buffer that is put into the hashtable. Meaning it is not possible to have NULL as value for any key (we even crash syslog-ng if the buffer could not be allocated). In short, you can use g_hash_table_lookup, there is no need for the extra safety of g_hash_table_lookup_extended.

Reply:
There is just one pitfall with RHEL6 which glib2 is missing `g_hash_table_lookup`. Any way to solve it?